src/baseline_bots/bots/smart_order_accepter_bot.py

```diff
@@ -24,6 +27,42 @@ class SmartOrderAccepterBot(RandomProposerBot):
     def __init__(self, power_name, game) -> None:
         super().__init__(power_name, game)
         self.alliance_props_sent = False
+        self.stance = stance_extraction.ScoreBasedStance()
+
+    def get_proposals(self, rcvd_messages: List[int, Message]) -> Dict[str, str]:
+        """
+        Extract proposal messages from received messages and checks for valid syntax before returning it
+        """
+        grammar = create_daide_grammar(level=130)
+        
+        # Extract messages containing PRP string
+        order_msgs = [msg[1] for msg in rcvd_messages if "PRP" in msg[1].message]
+
+        proposals = {}
+        for order_msg in order_msgs:
+            try:
+                # Parse using DAIDEPP functions
+                parse_tree = grammar.parse(order_msg.message)
+                output = daide_visitor.visit(parse_tree)
+
+                if output[0] == 'PRP':
+                    proposals[order_msg.sender] = order_msg.message
+            except Exception as e:
+                print(e)
+                pass
+
+        return proposals
+
+    def gen_pos_stance_messages(self, game_rec: json, ret_msgs: MessagesData) -> None:
```

Comment:
is json a Python type?

Reply:
I'll change this to Dict which seems more general and more appropriate

Reply:
But I am not sure what would be the type annotation for Dict's values since I had not worked on these types of game records.